app/controllers/api/integrations_controller.rb

```diff
@@ -30,13 +30,14 @@ def update
       proxy_pro_update
     elsif @proxy.save_and_deploy(proxy_params)
       environment = @proxy.service_mesh_integration? ? 'Production' : 'Staging'
+      api_backend = @proxy.api_backend
```

Comment:
Minor comment but imo better if this goes in the line after the `if @proxy.send_api_test_request!` because if it does not reach there, this is not needed so better not to execute stuff that is not needed